### PR TITLE
Use NsAs interface in Network Server instead of linking

### PR DESCRIPTION
**Pull request by rvolosatovs**
#### Summary

References https://github.com/TheThingsNetwork/lorawan-stack/issues/2312 #3263

#### Changes

- Remove `LinkApplication` from NS API
- Use `NsAs.HandleUplink`
- Only send FPort>0 uplinks to AS
- ~Optimize Redis task queue impl~ (see #3415 )

#### Testing

Unit tests, integration testing

##### Regressions

Application uplink flow

#### Notes for Reviewers

@adriansmares pushing downlinks to AS does not work for me, so it's quite difficult to test properly:
```
 DEBUG Finished unary call                      duration=18.953473ms error=rpc error: code = NotFound desc = error:pkg/redis:not_found (entity not found) error_correlation_id=c64c7144914c463692e8bf84f95e9444 error_name=not_found error_namespace=pkg/redis grpc_code=NotFound grpc_method=DownlinkQueuePush grpc_service=ttn.lorawan.v3.AppAs namespace=grpc
error:pkg/redis:not_found (entity not found)
```

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by rvolosatovs** [comment]
blocked by #3415

**Comment by rvolosatovs** [comment]
@adriansmares note that pushing of uplink to AS fails with the same error as OP as well, it is caused by this line https://github.com/rvolosatovs/lorawan-stack/blob/4675a4ac50982a6bceefe502ea0669328832dbbc/pkg/applicationserver/grpc.go#L80

**Review by johanstokking — COMMENTED** [review]
Please re-request review when this is rebased on Redis stuff.

**Comment by johanstokking** [comment]
What is the status here? Should I review? Should we wait for https://github.com/TheThingsNetwork/lorawan-stack/pull/3445?

**Comment by rvolosatovs** [comment]
> What is the status here? Should I review? Should we wait for #3445?

This is ready for review for 11 days already

**Comment by rvolosatovs** [comment]
We merge all PRs into #3263 and then merge the whole thing into 3.11

File: pkg/networkserver/grpc_asns.go [hunk 1]
```diff
@@ -65,72 +54,54 @@ func applicationJoinAcceptWithoutAppSKey(pld *ttnpb.ApplicationJoinAccept) *ttnp
...
-		return err
-	}
-	if err := rights.RequireApplication(ctx, ids, ttnpb.RIGHT_APPLICATION_LINK); err != nil {
-		return err
-	}
-
-	var wg sync.WaitGroup
-	wg.Add(1)
+func (ns *NetworkServer) processApplicationUplinkTask(ctx context.Context) error {
+	const retryInterval = time.Minute
```
Comment (johanstokking) [inline pkg/networkserver/grpc_asns.go]: Please declare const out of scope

File: pkg/networkserver/grpc_asns.go [hunk 2]
```diff
@@ -65,72 +54,54 @@ func applicationJoinAcceptWithoutAppSKey(pld *ttnpb.ApplicationJoinAccept) *ttnp
...
-				EndDeviceIdentifiers: up.EndDeviceIdentifiers,
-				CorrelationIDs:       up.CorrelationIDs,
-				Up: &ttnpb.ApplicationUp_JoinAccept{
-					JoinAccept: applicationJoinAcceptWithoutAppSKey(pld.JoinAccept),
-				},
-			}))
+		cl := ttnpb.NewNsAsClient(conn)
+		for ok := false; !ok; {
+			var sendErr bool
+			ok, err = f(100, func(ups ...*ttnpb.ApplicationUp) error {
```
Comment (johanstokking) [inline pkg/networkserver/grpc_asns.go]: 100 looks like a magic number

**Review by johanstokking — COMMENTED** [review]
Looks great!

@adriansmares can you review too please?

**Review by adriansmares — APPROVED** [review]

**Pull request merged**